lib/OpenQA/Scheduler/Model/Jobs.pm

```diff
@@ -256,7 +256,7 @@ sub schedule {
             for my $job (@jobs) {
                 try {
                     # Remove the associated worker and be sure to be in scheduled state.
-                    die "failed reset" unless $job->reschedule_state;
+                    $job->reschedule_state;
```

Comment:
so there is no useful return code anymore?

Reply:
There was no useful return code before because the function throws an exception in the error case. (And yes, the function was written as if it would return `0` when the DB operation failed. But that code was useless because possible exceptions from DBIx aren't handled.)